### ISSUE TITLE
GLA Bomb Truck can disguise into USA Spy Drone

### ISSUE THREAD

**Issue by xezon** [xezon/GeneralsGamePatch#204]
GLA Bomb Truck can disguise into USA Spy Drone.

**Comment by Jundiyy** [comment]
Think it's fine.

**Comment by ReLaX82** [comment]
Any ground vehicle is fine for me.

**Comment by xezon** [comment]
From survey comments:

> bomb trucks can mimick drones from usa making it almost impossible to hit them with forced fire or seeing them coming

**Comment by ImTimK** [comment]
Should almost stay because it's funny, but no, I think it's broken and unrealistic.

**Comment by commy2** [comment]
It can also disguise into a Battle Ship.

**Comment by kABUSE1** [comment]
Best way to fix this is keep it, but make them easy to hit with force-fire by increasing click hitbox. I am not sure that is feasible as I am quite sure that this would also change the hitbox for actual drones and an increased hitbox for them might have negative consequences, as they could overlap the vehicle they are following when you are trying to hit the vehicle and not the drone.

**Comment by xezon** [comment]
I think the best way to fix it is preventing disguise if target vehicle geometry is too small or too large, so that it cannot disguise as Battleship or Slave Drone.

**Comment by commy2** [comment]
Nah, `StealthUpdate` needs to have some `DisguiseForbiddenKindOf` and `DisguiseRequiredKindOf` lists and the ability should check against those. That's how the majority of abilities control what they can and can not target.
